File: Task-3/Five-in-One.js
// Function to get the maximum number in the array
function getMaxNumber(arr) {
  let max = arr[0];
  for (let i = 1; i < arr.length; i++) {
    if (arr[i] > max) {
      max = arr[i];
    }
  }
  return max;
}

// Function to get the minimum number in the array
function getMinNumber(arr) {
  let min = arr[0];
  for (let i = 1; i < arr.length; i++) {
    if (arr[i] < min) {
      min = arr[i];
    }
  }
  return min;
}

// Function to check if a number is prime
function isPrime(num) {
  if (num <= 1) {
    return false;
  }
  for (let i = 2; i <= Math.sqrt(num); i++) {
    if (num % i === 0) {
      return false;
    }
  }
  return true;
}

// Function to count prime numbers in the array
function countPrimes(arr) {
  let count = 0;
  for (const num of arr) {
    if (isPrime(num)) {
      count++;
    }
  }
  return count;
}

// Function to check if a number is a palindrome
function isPalindrome(num) {
  const numStr = num.toString();
  return numStr === numStr.split("").reverse().join("");
}

// Function to count palindrome numbers in the array
function countPalindromes(arr) {
  let count = 0;
  for (const num of arr) {
    if (isPalindrome(num)) {
      count++;
    }
  }
  return count;
}

// Function to get the number with the maximum number of divisors
function getMaxDivisorsNumber(arr) {
  function countDivisors(num) {
    let count = 0;
    for (let i = 1; i <= num; i++) {
      if (num % i === 0) {
        count++;
      }
    }
    return count;
  }

  let maxDivisors = 0;
  let numberWithMaxDivisors;

  for (const num of arr) {
    const divisorsCount = countDivisors(num);
    if (divisorsCount > maxDivisors) {
      maxDivisors = divisorsCount;
      numberWithMaxDivisors = num;
    }
  }

  return numberWithMaxDivisors;
}

// Function to process input string
function processInput(input) {
  const lines = input.trim().split("\n");
  const N = parseInt(lines[0]);
  const arr = lines[1].split(" ").map(Number);

  console.log(`The maximum number : ${getMaxNumber(arr)}`);
  console.log(`The minimum number : ${getMinNumber(arr)}`);
  console.log(`The number of prime numbers : ${countPrimes(arr)}`);
  console.log(`The number of palindrome numbers : ${countPalindromes(arr)}`);
  console.log(
    `The number that has the maximum number of divisors : ${getMaxDivisorsNumber(
      arr
    )}`
  );
}

// Example usage
const input = "5\n8 2 14 1 83";
processInput(input);

File: Task-3/Five in One.c
#include <stdio.h>

// Function to check if a number is prime
int isPrime(int num) {
    if (num < 2) {
        return 0;
    }

    for (int i = 2; i * i <= num; i++) {
        if (num % i == 0) {
            return 0;
        }
    }

    return 1;
}

// Function to check if a number is a palindrome
int isPalindrome(int num) {
    int original = num, reversed = 0;

    while (num > 0) {
        reversed = reversed * 10 + num % 10;
        num /= 10;
    }

    return original == reversed;
}

// Function to count the number of divisors of a number
int countDivisors(int num) {
    int count = 0;

    for (int i = 1; i <= num; i++) {
        if (num % i == 0) {
            count++;
        }
    }

    return count;
}

int main() {
    int N;
    scanf("%d", &N);

    int A[100];

    // Read the array
    for (int i = 0; i < N; i++) {
        scanf("%d", &A[i]);
    }

    // Find maximum and minimum
    int maxNum = A[0], minNum = A[0];

    for (int i = 1; i < N; i++) {
        if (A[i] > maxNum) {
            maxNum = A[i];
        }
        if (A[i] < minNum) {
            minNum = A[i];
        }
    }

    // Count primes, palindromes, and divisors
    int primeCount = 0, palindromeCount = 0, maxDivisors = 0, maxDivisorNum = A[0];

    for (int i = 0; i < N; i++) {
        if (isPrime(A[i])) {
            primeCount++;
        }
        if (isPalindrome(A[i])) {
            palindromeCount++;
        }
        int divisors = countDivisors(A[i]);
        if (divisors > maxDivisors || (divisors == maxDivisors && A[i] > maxDivisorNum)) {
            maxDivisors = divisors;
            maxDivisorNum = A[i];
        }
    }

    // Print results
    printf("The maximum number : %d\n", maxNum);
    printf("The minimum number : %d\n", minNum);
    printf("The number of prime numbers : %d\n", primeCount);
    printf("The number of palindrome numbers : %d\n", palindromeCount);
    printf("The number that has the maximum number of divisors : %d\n", maxDivisorNum);

    return 0;
}
